src/signal.rs

```diff
@@ -499,4 +505,89 @@ mod tests {
         // forever as the loop keeps running. Since we don't join, the thread will become detached
         // as the handle is dropped, and will be killed when the process/main thread exits.
     }
+
+    #[test]
+    fn test_get_blocked_signals() {
+        let mut sigset: sigset_t = unsafe { mem::zeroed() };
+
+        // Block a signal.
+        unsafe {
+            sigemptyset(&mut sigset);
+            sigaddset(&mut sigset, SIGRTMIN() + 0);
+            pthread_sigmask(SIG_BLOCK, &sigset, null_mut());
+        }
+
+        assert!(get_blocked_signals().unwrap().contains(&(SIGRTMIN() + 0)));
```

Comment:
Can we test `get_blocked_signals` as part of the next function (where we also block some signal), and remove this one altogether?

Reply:
Sure. This is a good idea. I will just keep the function to be tested in unblock_signal.